lib/modules/manager/sbt/extract.ts

```diff
@@ -335,11 +385,121 @@ export function extractPackageFile(
     return null;
   }
 
-  const { deps, packageFileVersion } = parsedResult;
+  return parsedResult;
+}
 
-  if (!deps.length) {
-    return null;
+function prepareLoadPackageFiles(
+  packageFilesContent: { packageFile: string; content: string }[]
+): {
+  globalVars: ParseOptions['globalVars'];
+  registryUrls: string[];
+  scalaVersion: ParseOptions['scalaVersion'];
+} {
+  // Return variable
+  let globalVars: Variables = {};
+  let registryUrls: string[] = [REGISTRY_URLS.mavenCentral];
+  let scalaVersion: string | undefined = undefined;
+
+  for (const { packageFile, content } of packageFilesContent) {
+    const acc: PackageFile & ParseOptions = {
+      deps: [],
+      registryUrls,
+      localVars: globalVars,
+      globalVars: {},
+      packageFile,
+    };
+    const res = extractPackageFile(content, acc);
+
+    if (res) {
+      globalVars = { ...globalVars, ...res.localVars };
+      if (res.registryUrls) {
+        registryUrls = [...registryUrls, ...new Set(res.registryUrls)];
+      }
+      if (res.scalaVersion) {
+        scalaVersion = res.scalaVersion;
+      }
+    }
+  }
+
+  return {
+    globalVars,
+    registryUrls,
+    scalaVersion,
+  };
+}
+
+export async function extractAllPackageFiles(
+  _config: ExtractConfig,
+  packageFiles: string[]
+): Promise<PackageFile[] | null> {
+  // Read packages and store in groupPackageFileContent
+  // group package file by its folder
+  const groupPackageFileContent: GroupFilenameContent = {};
+  for (const packageFile of packageFiles) {
+    const content = await readLocalFile(packageFile, 'utf8');
+    if (!content) {
+      logger.trace({ packageFile }, 'packageFile has no content');
+      continue;
+    }
+    const group = upath.dirname(packageFile);
+    groupPackageFileContent[group] ??= [];
+    groupPackageFileContent[group].push({ packageFile, content });
+  }
+
+  // 1. globalVariables from project/ and root package file
+  // 2. registry from all package file
+  // 3. Project's scalaVersion - use in parseDepExpr to add suffix eg. "_2.13"
+  const { globalVars, registryUrls, scalaVersion } = prepareLoadPackageFiles([
+    ...(groupPackageFileContent['project'] ?? []), // in project/ folder
+    ...(groupPackageFileContent['.'] ?? []), // root
+  ]);
+
+  const mapDepsToPackageFile: Record<string, PackageDependency[]> = {};
+  // Start extract all package files
+  for (const packageFileContents of Object.values(groupPackageFileContent)) {
+    // Extract package file by its group
+    // local variable is share within its group
+    for (const { packageFile, content } of packageFileContents) {
+      const res = extractPackageFile(content, {
```

Comment:
This will parse at least all files of `groupPackageFileContent['project']` and `groupPackageFileContent['.']` again. It's not clear to me why this would be necessary?!

Reply:
This is to store global variables. before actually extract all package files

For example: in file `./build.sbt` try to resolve the variable stored in `project/Dependencies.scala`